### PR TITLE
Call Home Enabled in console

### PR DESCRIPTION
**Pull request by bexsoft**
## What does this do?

Enabled Call Home functionality in Console

## How does it look?

Screenshot 2023-02-09 at 20 16 37
Screenshot 2023-02-09 at 20 09 12
Screenshot 2023-02-09 at 20 08 55
Screenshot 2023-02-09 at 20 08 51
Screenshot 2023-02-09 at 20 08 45
Screenshot 2023-02-09 at 20 08 38
Screenshot 2023-02-09 at 20 08 34

Signed-off-by: Benjamin Perez <benjamin@bexsoft.net>

### CODE REVIEW

**Review by jinapurapu — CHANGES_REQUESTED** [review]

File: portal-ui/src/screens/Console/Support/CallHomeConfirmation.tsx [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+      })
+      .catch((err: ErrorResponseHandler) => {
+        setLoading(false);
+        dispatch(setErrorSnackMessage(err));
+      });
+  };
+
+  return (
+    <ConfirmDialog
+      title={currentStatus ? "Disable CallHome" : "Enable CallHome"}
```
Comment (jinapurapu) [inline portal-ui/src/screens/Console/Support/CallHomeConfirmation.tsx:59]:
```suggestion
      title={currentStatus ? "Disable Call Home" : "Enable Call Home"}
```

**Review by jinapurapu — CHANGES_REQUESTED** [review]

**Comment by jinapurapu** [comment]
I don't have Subnet credentials to test the API, but UI LGTM

**Comment by cniackz** [comment]
@bexsoft when you get a chance, please resolve the conflicts:

Screen Shot 2022-10-07 at 2 50 27 PM

**Comment by cniackz** [comment]
ok ignore my previous comment, maybe we need to have this on subnet first...

**Comment by bexsoft** [comment]
> @bexsoft when you get a chance, please resolve the conflicts:
> 
> Screen Shot 2022-10-07 at 2 50 27 PM

Fixed

**Comment by bexsoft** [comment]
Original PR: https://github.com/minio/console/pull/1664

**Comment by allanrogerr** [comment]
@bexsoft How do I enable these new options on the console - again I pulled the branch, checked out, ran `make k8sdev` and reloaded the console pod. I cannot see Call Home.

**Review by dvaldivia — APPROVED** [review]

**Comment by prakashsvmx** [comment]
@bexsoft @dvaldivia 

the call home can be enabled selectively as well
e.g
 `mc support callhome enable <alias>   --logs ` 
 `mc support callhome enable <alias>   --diag `

So we would have to add these options in the UI as well. 
Could you please check this ?

**Comment by bexsoft** [comment]
> @bexsoft @dvaldivia
> 
> the call home can be enabled selectively as well e.g `mc support callhome enable <alias> --logs ` `mc support callhome enable <alias> --diag `
> 
> So we would have to add these options in the UI as well. Could you please check this ?

Hi @prakashsvmx !

Yes, Let me add them, this is an old PR so these features were not available at the time of development.

**Review by allanrogerr — APPROVED** [review]
Tested

**Review by allanrogerr — CHANGES_REQUESTED** [review]
After attempting to use the API, the changes are not visible in the UI after refreshing the page

➜  portal-ui git:(call-home-api) ✗ sudo mc support callhome disable myminio --logs --insecure
Logs is now disabled

but,

image

**Comment by bexsoft** [comment]
> After attempting to use the API, the changes are not visible in the UI after refreshing the page
> 
> ➜ portal-ui git:(call-home-api) ✗ sudo mc support callhome disable myminio --logs --insecure Logs is now disabled
> 
> but,
> 
> image

Yes, this was related to a functionality that was not implemented at the time this PR was originally made, implemented new functionality and redesigned screens

**Review by dvaldivia — DISMISSED** [review]

**Comment by pjuarezd** [comment]
Tested, build console and minio and loaded it into k8s kind using docker image, now the UI shows the callhome section

But keep receiving error code 500 in endpoint `https://localhost:9444/api/v1/support/callhome`.

```json
{"code":500,"detailedMessage":"there is no target `subnet` for subsystem `logger_webhook`","message":"an error occurred, please try again"}
```

Could I be missing something?

Screenshot 2023-02-09 at 8 26 07 PM

So, I though it could be that I am missing to login  in subnet, or register the cluster, so I went to subnet and reguistered it (using airgap method)

Screenshot 2023-02-09 at 8 36 40 PM

```sh
 ~ mc license update callhome ./cluster-license-.key --insecure
License updated successfully for callhome
```

And I still see the "Register your cluster" button, additionally to the redbanner error


Screenshot 2023-02-09 at 8 17 19 PM

Screenshot 2023-02-09 at 8 45 38 PM

So, noticed that a restart of the minio services is required to show the cluster as registered, so perhaps is worth show the "The instance needs to be restarted for configuration changes to take effec" banner?

Screenshot 2023-02-09 at 8 55 39 PM

Anyways, I restarted the cluster like this

```~ mc admin service restart callhome --insecure
Restart command successfully sent to `callhome`. Type Ctrl-C to quit or wait to follow the status of the restart process.
...!!
Restarted `callhome` successfully in 3 seconds
```

But I keep receiving that red banner error message, but at least I can enable callhome now

Screenshot 2023-02-09 at 9 03 43 PM


A console hard refresh and now the red banner is gone, looks like console (or minio, not sure) is looking for that webhook  even before callhome is activated, because once calhome is activated it shows the logging webhook  and the red banner dissapeared.
Screenshot 2023-02-09 at 9 08 56 PM

**Comment by bexsoft** [comment]
Fixing red banner

File: portal-ui/src/screens/Console/Support/CallHome.tsx [hunk 2]
```diff
@@ -0,0 +1,328 @@
...
+
+  const disableCallHomeAction = () => {
+    setDisableMode(true);
+    setShowConfirmation(true);
+  };
+
+  let mainVariant: "regular" | "callAction" = "regular";
+
+  if (
+    /*clusterRegistered &&*/ diagEnabled !== oDiagEnabled ||
```
Comment (jinapurapu) [inline portal-ui/src/screens/Console/Support/CallHome.tsx:101]: Please remove comment

File: portal-ui/src/screens/Console/Support/CallHome.tsx [hunk 3]
```diff
@@ -0,0 +1,328 @@
...
+                      setLogsEnabled(event.target.checked);
+                    }}
+                    label={"Send Logs information"}
+                    disabled={!clusterRegistered}
+                  />
+                  <PromoLabels
+                    title={"When you enable logs"}
+                    text={
+                      "Streamline your log management with MinIO's Subnet! Upload your logs and easily access them from your\n" +
+                      "                  subnet account. Get real-time insights and troubleshoot issues quickly"
```
Comment (jinapurapu) [inline portal-ui/src/screens/Console/Support/CallHome.tsx:205]:
```suggestion
                      "                  subnet account. Get real-time insights and troubleshoot issues quickly."
```

File: portal-ui/src/screens/Console/Support/CallHome.tsx [hunk 4]
```diff
@@ -0,0 +1,328 @@
...
+                      setDiagEnabled(event.target.checked);
+                    }}
+                    label={"Send Diagnostics information"}
+                    disabled={!clusterRegistered}
+                  />
+                  <PromoLabels
+                    title={"When you enable diagnostics"}
+                    text={
+                      "Understand your storage with MinIO and unlock the potential of efficient health diagnostics. Ensure\n" +
+                      "                  seamless performance and identify potential issues before they escalate"
```
Comment (jinapurapu) [inline portal-ui/src/screens/Console/Support/CallHome.tsx:185]:
```suggestion
                      "                  seamless performance and identify potential issues before they escalate."
```

File: portal-ui/src/screens/Console/Support/CallHomeConfirmation.tsx [hunk 5]
```diff
@@ -0,0 +1,218 @@
...
+  return (
+    <ModalWrapper
+      modalOpen={open}
+      title={disable ? "Disable Call Home" : "Edit Call Home Configurations"}
+      onClose={() => onClose(false)}
+      titleIcon={<CallHomeMenuIcon />}
+    >
+      {disable ? (
+        <Fragment>
+          Please Acknowledge that doing this action, we will not receive updated
```
Comment (jinapurapu) [inline portal-ui/src/screens/Console/Support/CallHomeConfirmation.tsx:75]:
```suggestion
          Please Acknowledge that after doing this action, we will no longer receive updated
```

File: portal-ui/src/screens/Console/Support/CallHomeConfirmation.tsx [hunk 6]
```diff
@@ -0,0 +1,218 @@
...
+          cluster information automatically anymore, losing the potential
```
Comment (jinapurapu) [inline portal-ui/src/screens/Console/Support/CallHomeConfirmation.tsx:76]:
```suggestion
          cluster information automatically, losing the potential
```

**Review by jinapurapu — CHANGES_REQUESTED** [review]

File: portal-ui/src/screens/Console/Support/CallHome.tsx (hunk 2 above)
Comment (bexsoft) [reply portal-ui/src/screens/Console/Support/CallHome.tsx]: Fixed

File: portal-ui/src/screens/Console/Support/CallHome.tsx [hunk 7]
```diff
@@ -0,0 +1,328 @@
...
+                      setLogsEnabled(event.target.checked);
+                    }}
+                    label={"Send Logs information"}
+                    disabled={!clusterRegistered}
+                  />
+                  <PromoLabels
+                    title={"When you enable logs"}
+                    text={
+                      "Streamline your log management with MinIO's Subnet! Upload your logs and easily access them from your\n" +
+                      "                  subnet account. Get real-time insights and troubleshoot issues quickly."
```
Comment (jinapurapu) [inline portal-ui/src/screens/Console/Support/CallHome.tsx:205]:
```suggestion
                      "                  SUBNET account. Get real-time insights and troubleshoot issues quickly."
```

**Review by jinapurapu — CHANGES_REQUESTED** [review]

**Review by jinapurapu — APPROVED** [review]

**Review by allanrogerr — APPROVED** [review]

**Review by pjuarezd — APPROVED** [review]
no red banner, all good!

**Pull request merged**